bin/i18n-update.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+# 
+# Handles the process of updating the i18n localizations of plugins, including Gutenberg.
+# The main goals of this command are:
+#
+#   1. Download translation files and optimize them by filtering out the unused strings,
+#      previously extracted from the source map files of React Native bundle.
+#
+#      This step produces the following output files:
+#      - src/i18n-cache/{PLUGIN_NAME}/data/{LOCALE}.json     [Translation files]
+#      - src/i18n-cache/{PLUGIN_NAME}/data/index.js          [JS file to import translations]
+#
+#   2. Generate localization strings files that include the strings only used in native JS
+#      source code files ("native.js", "ios.js or "android.js"). The translations of these
+#      strings are not included in the GlotPress projects of their plugins hence, they require
+#      to be requested as part of the main apps’ strings.
+#
+#      This step produces the following output files:
+#      - bundle/android/strings.xml                     [Localization strings files for Android platform]
+#      - bundle/ios/GutenbergNativeTranslations.swift   [Localization strings files for iOS platform]
+
+# Exit if any command fails
+set -e
```

Comment:
Nitpick: might want to consider using `set -euxo pipefail` see https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/

`set -u` is a bit tricky. just need to use the `INPUT=${VAR:-"some default"}` form to make sure everything is set.

Reply:
It's been awhile since I've read that article 😊  Probably don't want `set -x` in there once this is ready to ship but it helps for debugging.

Reply:
I applied the suggestion in https://github.com/wordpress-mobile/gutenberg-mobile/pull/4366/commits/8bc16298d8302cd8052e1cb59333def57759d97a.